fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashActivity.java

```diff
@@ -233,59 +101,83 @@ public static DigitalCashViewModel obtainViewModel(FragmentActivity activity) {
+              Log.d(TAG, "Opening history fragment");
+              openHistoryFragment();
+            } else if (id == R.id.digital_cash_send_menu) {
+              Log.d(TAG, "Opening send fragment");
+              openSendFragment();
+            } else if (id == R.id.digital_cash_receive_menu) {
+              Log.d(TAG, "Opening receive fragment");
+              openReceiveFragment();
+            } else if (id == R.id.digital_cash_issue_menu) {
+              Log.d(TAG, "Trying to open issue fragment");
+              handleOpenIssue();
+            }
           }
           return true;
         });
   }
 
-  public void setupFragment(int id) {
-    if (id == R.id.fragment_digital_cash_home) {
-      setCurrentFragment(
-          getSupportFragmentManager(),
-          R.id.fragment_digital_cash_home,
-          DigitalCashHomeFragment::newInstance);
-    } else if (id == R.id.fragment_digital_cash_history) {
-      setCurrentFragment(
-          getSupportFragmentManager(),
-          R.id.fragment_digital_cash_history,
-          DigitalCashHistoryFragment::newInstance);
-    } else if (id == R.id.fragment_digital_cash_send) {
-      setCurrentFragment(
-          getSupportFragmentManager(),
-          R.id.fragment_digital_cash_send,
-          DigitalCashSendFragment::newInstance);
-    } else if (id == R.id.fragment_digital_cash_receive) {
-      setCurrentFragment(
-          getSupportFragmentManager(),
-          R.id.fragment_digital_cash_receive,
-          DigitalCashReceiveFragment::newInstance);
-    } else if (id == R.id.fragment_digital_cash_issue) {
-      setCurrentFragment(
-          getSupportFragmentManager(),
-          R.id.fragment_digital_cash_issue,
-          DigitalCashIssueFragment::newInstance);
-    } else if (id == R.id.fragment_digital_cash_receipt) {
-      setCurrentFragment(
-          getSupportFragmentManager(),
-          R.id.fragment_digital_cash_receipt,
-          DigitalCashReceiptFragment::newInstance);
+  private void openHomeFragment() {
+    setCurrentFragment(
+        getSupportFragmentManager(),
+        R.id.fragment_digital_cash_home,
+        DigitalCashHomeFragment::newInstance);
+  }
+
+  private void openHistoryFragment() {
+    setCurrentFragment(
+        getSupportFragmentManager(),
+        R.id.fragment_digital_cash_history,
+        DigitalCashHistoryFragment::newInstance);
+  }
+
+  private void openSendFragment() {
+    setCurrentFragment(
+        getSupportFragmentManager(),
+        R.id.fragment_digital_cash_send,
+        DigitalCashSendFragment::newInstance);
+  }
+
+  private void openReceiveFragment() {
+    setCurrentFragment(
+        getSupportFragmentManager(),
+        R.id.fragment_digital_cash_receive,
+        DigitalCashReceiveFragment::newInstance);
+  }
+
+  private void handleOpenIssue() {
+    if (!mViewModel
+        .getCurrentLao()
+        .getValue()
+        .getOrganizer()
+        .equals(mViewModel.getKeyManager().getMainPublicKey())) {
```

Comment:
it might be more readable / expressive to use intermediate variables.

pseudo-code:
```
    organizerKey = mViewModel.getCurrentLao().getValue().getOrganizer()
    myKey = mViewModel.getKeyManager().getMainPublicKey()

    if (!myKey.equals(organizerKey)) {
```

Reply:
Agreed !